src/server/usage/classes/collector_set.js

```diff
@@ -92,6 +92,25 @@ export class CollectorSet {
     });
   }
 
+  bulkFormat(data, collectors = this._collectors) {
+    if (!Array.isArray(collectors)) {
+      throw new Error(`bulkFormat method given bad collectors parameter: ` + typeof collectors);
+    }
+
+    return data.reduce((accum, collectedData) => {
+      if (isEmpty(collectedData)) {
+        return accum;
+      }
+      const collector = collectors.find(_c => _c.type === collectedData.type);
```

Comment:
This looks like it can use the `getCollectorByType` method of a CollectorSet instance